src/main/java/org/mule/module/apikit/model/AMFWrapper.java

```diff
@@ -85,24 +87,18 @@ public class AMFWrapper {
   private static final String EDM_TIME_PATTERN =
       "^PT([0-1]?[0-9]|2[0-3])H[0-5][0-9]M(|[0-5][0-9]S)$";
 
-  private List<NodeShape> nodeShapesList;
-  private Map<String, NodeShape> shapes = new HashMap<>();
+  private List<NodeShape> nodeShapesList = new ArrayList<>();
   private EntityDefinitionSet entityDefinitionSet = new EntityDefinitionSet();
 
   public AMFWrapper(String ramlPath, Scheduler scheduler)
       throws OdataMetadataFormatException, OdataMetadataFieldsException {
-    try {
-      ExecutionEnvironment executionEnvironment =
-          scheduler != null ? new ExecutionEnvironment(scheduler) : new ExecutionEnvironment();
-      Environment environment = DefaultEnvironment.apply(executionEnvironment);
-      AMF.init(executionEnvironment).get();
-      initNodeShapesList(ramlPath, environment);
-    } catch (ExecutionException | InterruptedException e) {
-      throw new OdataMetadataFormatException("RAML is invalid. See log list.");
-    }
+
+    ExecutionEnvironment executionEnvironment =
+        scheduler != null ? new ExecutionEnvironment(scheduler) : new ExecutionEnvironment();
+
+    tryParse(ramlPath, executionEnvironment);
```

Comment:
I'm missing context here probably... Is this necessary because we use the same wrapper for both library and api? Can't we delegate the parsing to subclasses that specialize it?

Reply:
yes, we can check with if (path contains odata.raml) or something like that

Reply:
the thing here is that we need to re-write the test cases, and pray to dont break anything 😄

Reply:
in scaffolding time, we can assume that the library file is odata.raml , but in runtime, it can change

Reply:
Makes sense! It's not worth the effort then 👍